internal/juju/charms.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the Apache License, Version 2.0, see LICENCE file for details.
+
+package juju
+
+import (
+	"fmt"
+
+	"github.com/juju/juju/api"
+	"github.com/juju/names/v5"
+)
+
+type infoParameters struct {
+	Tag     string `json:"tag"`
+	Channel string `json:"channel,omitempty"`
+}
+
+type charmHubEntityInfoResult struct {
+	Result infoResponse          `json:"result"`
+	Errors charmhubErrorResponse `json:"errors"`
+}
+
+type charmhubErrorResponse struct {
+	Error charmhubError `json:"error-list"`
+}
+
+type charmhubError struct {
+	Code    string `json:"code"`
+	Message string `json:"message"`
+}
+
+type infoResponse struct {
+	ID    string         `json:"id"`
+	Name  string         `json:"name"`
+	Charm *charmHubCharm `json:"charm,omitempty"`
+}
+
+type charmHubCharm struct {
+	Subordinate bool `json:"subordinate"`
+}
+
+// IsSubordinateCharmParameters holds the parameters for the IsSubordinateCharm method.
+type IsSubordinateCharmParameters struct {
+	Name    string
+	Channel string
+}
+
+type charmsClient struct {
+	Connection api.Connection
+}
+
+// newCharmsClient creates a new charmsClient that works with Juju 2.9 controller to fetch
+// information on whether a charm is a subordinate charm.
```

Comment:
Please explain that for 3 onwards, the controller handles this? Or put that info somewhere obvious so it is super clear why we need this only in 2.9